x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/PostAnalyzer.java

```diff
@@ -47,21 +49,24 @@ public LogicalPlan postAnalyze(LogicalPlan plan, EqlConfiguration configuration)
             Holder<Boolean> hasJoin = new Holder<>(Boolean.FALSE);
 
             Source projectCtx = synthetic("<implicit-project>");
-            // first per KeyedFilter
-            plan = plan.transformUp(KeyedFilter.class, k -> {
+            if (plan.anyMatch(Sequence.class::isInstance)) {
```

Comment:
Because in a Sample there is no notion of limit (so far at least) and what this is doing is to add a `limit` plan to the tree.

Reply:
`fetch_size` is different. What you probably meant is [`size`](https://www.elastic.co/guide/en/elasticsearch/reference/8.2/eql-search-api.html#eql-search-api-request-body). And yes, for now, there is no limit and it will return all keys.

I will add `fetch_size` support in a new commit. This one is about the size of the results page returned by the `composite` aggregation and it has a very similar meaning in sequences as well.

Reply:
yes, I had `size` in mind. Isn't support for `size` somewhat crucial? Otherwise a query might easily OOM due to too many samples.

Reply:
+1 on addressing limit/size in a separate PR if it's non-simplistic - this one is already quite big.
Potentially open a separate branch from sample to avoid any impact on releases.